driver-core/src/main/com/mongodb/operation/AggregateOperationImpl.java

```diff
@@ -228,7 +228,8 @@ private BsonDocument getCommand(final ConnectionDescription description, final S
         appendReadConcernToCommand(sessionContext, commandDocument);
         commandDocument.put("pipeline", pipelineCreator.create(description, sessionContext));
         if (maxTimeMS > 0) {
-            commandDocument.put("maxTimeMS", new BsonInt64(maxTimeMS));
+            commandDocument.put("maxTimeMS", maxTimeMS > Integer.MAX_VALUE
```

Comment:
Necessary because the tests expect a BSON Int32.

Reply:
Urgh - not nice.